nativelib/src/main/scala-3/scala/scalanative/runtime/Continuations.scala

```diff
@@ -0,0 +1,168 @@
+package scala.scalanative.runtime
+
+import scala.scalanative.unsafe.*
+import scala.scalanative.runtime.libc.{malloc, free}
+import scala.scalanative.runtime.Intrinsics.*
+import scala.collection.mutable
+
+import scala.util.Try
+import scala.scalanative.meta.LinktimeInfo.isWindows
+
+object Continuations:
+  import ContImpl.*
+
+  opaque type BoundaryLabel[-T] = ContImpl.BoundaryLabel
+
+  ContImpl.init(CFuncPtr2.fromScalaFunction(allocateBlob))
+
+  /** Marks the given body as suspendable with a `BoundaryLabel` that `suspend`
+   *  can refer to. Forwards the return value of `body`, or the return value of
+   *  the `suspend` call, if it happens during the execution of `f`.
+   *
+   *  __Safety__: the passed-in `BoundaryLabel` cannot be used outside of the
+   *  scope of the body. Suspending to a `BoundaryLabel` not created by a
+   *  `boundary` call higher on the same call stack is undefined behaviour.
+   */
+  inline def boundary[T](inline body: BoundaryLabel[T] ?=> T): T =
+    if isWindows then UnsupportedFeature.continuations()
+    val call: ContFnT = (x: ContImpl.BoundaryLabel) =>
+      objToPtr(Try(body(using x)))
+    val resP = ContImpl.boundary(contFn, objToPtr(call))
+    objFromPtr(resP).asInstanceOf[Try[T]].get
```

Comment:
Lots of `objToPtr`/`objFromPtr` can be safely omited. 
If we would define `type ContFnT[T] = ContImpl.BoundaryLabel => Try[T]` and would pass it as following, we would get exactly the same result, with a cleaner and more readable code. Any reference type passed to extern function is passed as a pointer to itself. 

```suggestion
    val call: ContFnT = (x: ContImpl.BoundaryLabel) => Try(body(using x))
    ContImpl.boundary(contFn, call).get
```

It might require a few more type annotation, but eventually we could achive more type-safe and readable code. 
If we'd go that way the only issue would be the `CFuncPtr2` types which cannot be generic but these can be easily adapted as follows: 
```Scala 
  type ContFn[T] = CFuncPtr2[BoundaryLabel, ContFnT[T], Try[T]]
  type ContFnT[T] = ContImpl.BoundaryLabel => Try[T] // probably sume different name would be usefull 
  private val contFnAny: ContFn[Any] = CFuncPtr2.fromScalaFunction{ (label, fp) => fp(label) } // general
  inline def contFn[T] = contFnAny.asInstanceOf[ContFn[T]] // specific

```

Reply:
Thanks! I did not know this was possible. 

I did the same thing for both `suspend` and `resume`. Things look much better!